app/models/user_vote.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+class UserVote < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :poll
+
+  validates :user_id, uniqueness: { scope: :poll_id }
```

Comment:
Or you can make this a DB-level constraint (_composite unique index_) and avoid an additional DB query for every `save!` for this class. (You'll get database errors instead of validation one, though.)